obal/data/playbooks/library/rpmspec_changelog_check.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+from ansible.module_utils.basic import AnsibleModule
+
+import os
+import subprocess
+import glob
+
+def run_module():
+    module_args = dict(
+        directory=dict(type='str', required=True),
+    )
+
+    result = dict(
+        changed=False,
+        changelog_version_release='',
+        specfile_version_release=''
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    if module.check_mode:
+        return result
+
+    specfile = glob.glob(os.path.join(module.params['directory'], "*.spec"))[0]
+
+    cmd = ['rpm', '--query', '--changelog', '--specfile', specfile]
+    result['changelog_version_release'] = subprocess.check_output(cmd).split("\n")[0].split(" ")[-1]
+
+    cmd = ['rpm', '--query', '--queryformat', '%{version}-%{release}\n', '--undefine', 'dist', '--specfile', specfile]
+    result['specfile_version_release'] = subprocess.check_output(cmd).split("\n")[0]
+
+    if result['changelog_version_release'] != result['specfile_version_release']:
+        module.fail_json(msg="changelog has not been updated for the current version-release", **result)
```

Comment:
I feel that most developers would not fully know what to do with this statement, and that we can potentially lead them in the right direction with something like:

"changelog entry missing for %s" % result['specfile_version_release']

Reply:
updated